src/ScriptBuilder/CSharpCounterWriter.cs

```diff
@@ -13,23 +13,28 @@ public static void WriteCode(string scriptPath, IEnumerable<DurationDefinition>
             {
                 var stringBuilder = new StringBuilder();
 
-                var slaCounterDefinition = @"new CounterCreationData(""SLA violation countdown"", ""Seconds until the SLA for this endpoint is breached."", PerformanceCounterType.NumberOfItems32),";
+                var slaCounterDefinition = $@"new CounterCreationData(""SLA violation countdown"", ""{CounterNameAndDescriptionConventions.LegacySlaViolationCounterDescription}"", PerformanceCounterType.NumberOfItems32),";
                 stringBuilder.AppendLine(slaCounterDefinition.PadLeft(slaCounterDefinition.Length + 8));
 
                 foreach (var duration in durations)
                 {
                     var averageTimerName = duration.Name.GetAverageTimerCounterName();
                     var averageTimerBase = duration.Name.GetAverageTimerBaseCounterName();
+                    var averateTimerDescription = CounterNameAndDescriptionConventions.GetAverageTimerDescription(duration.Name, duration.Description);
 
-                    var durationAverageDefinition = $@"new CounterCreationData(""{averageTimerName}"", ""{duration.Description}"", PerformanceCounterType.AverageTimer32),";
+                    var durationAverageDefinition = $@"new CounterCreationData(""{averageTimerName}"", ""{averateTimerDescription}"", PerformanceCounterType.AverageTimer32),";
```

Comment:
AverageTimer32 might not be sufficient. Ticks is a fairly big number so what if you configure a concurrency of 1000+ ? For example, the stopwatch frequency on my computer is 3914062, AverageTimer32 max size is 2147483648 meaning the max duration to express is 2147483648/3914062 = ~548 seconds or  ~9 minutes which I think is not enough so it must be a *AverageTimer64*